source_control/gh_keys.py

```diff
@@ -0,0 +1,177 @@
+
+# Gets a single public key (authetication required)
+- gh_keys: action=get_key user=leocomelli passwd=secret key_id=8767854
+
+# Adds a new public key (authentication required)
+- gh_keys: action=add_key user=leocomelli passwd=secret title=my_new_key key=/home/leocomelli/.ssh/id_rsa.pub
+
+# Removes an existing ssh key
+- gh_keys: action=remove_key user=leocomelli passwd=secret key_id=8767854
+'''
+
+GH_API_URL = "https://api.github.com/%s"
+
+class GHKeys(object):
+
+  def __init__(self, module):
+    self.action = module.params['action']
+    self.user   = module.params['user']
+    self.passwd = module.params['passwd']
+    self.title  = module.params['title']
+    self.key    = module.params['key']
+    self.key_id = module.params['key_id']
+  
+  def work(self):
+    return {
+      'list_keys'  : self.list_keys,
+      'get_key'    : self.get_key,
+      'add_key'    : self.add_key,
+      'remove_key' : self.remove_key
+    }[self.action]()
+
+  def list_keys(self):
+    if self.passwd is None:
+      url = GH_API_URL % "users/%s/keys" % self.user
+      response = requests.get(url)  
+    else:
+      url = GH_API_URL % "user/keys"
+      response = requests.get(url, auth=(self.user, self.passwd))  
+
+    return response.text
+
+  def get_key(self):
+    self.validate_fileds('get_key', ['key_id', 'passwd'])    
+
+    url = GH_API_URL %  "user/keys/%s" % self.key_id
+    response = requests.get(url, auth=(self.user, self.passwd)) 
+
+    return response.text
+
+  def add_key(self):
+    self.validate_fileds('add_key', ['title', 'key', 'passwd'])
+
+    with open(self.key, 'r') as content_file:
+      content = content_file.read()
+
+    url = GH_API_URL %  "user/keys"
+    data = json.dumps({ 'title' : self.title, 'key' : content })
+    response = requests.post(url, auth = (self.user, self.passwd), data = data) 
+
+    return response.text
+
+  def remove_key(self):
+    self.validate_fileds('remove_key', ['key_id', 'passwd'])
+
+    url = GH_API_URL %  "user/keys/%s" % self.key_id
+    response = requests.delete(url, auth=(self.user, self.passwd)) 
+
+    return response.text
+
+  def validate_fileds(self, action, fields):
+    for field in fields:
+      if getattr(self, field) is None:
+        raise ValueError(field + " cannot be null for action [" + action + "]")
+
+
+def convert_bool_to_str(value):
+  return value.replace('true', '"true"').replace('false', '"false"')
+
+def main():
+  module = AnsibleModule(
+    argument_spec = dict(
+      passwd = dict(),
+      title  = dict(),
+      key    = dict(),
+      key_id = dict(),
+      user   = dict(required=True),
+      action = dict(required=True, choices=['list_keys', 'get_key', 'add_key', 'remove_key']),
+    )
+  )  
+
+  gh_keys = GHKeys(module)
+  try:
+    result = gh_keys.work()
+    if 'message' in result:
+      raise RuntimeError(result)
+
+    module.exit_json(changed=True, stdout=result)
+  except (RuntimeError, ValueError) as err:
```

Comment:
Using the `as` syntax is not compatible with python 2.4, you will need to use a `,` instead.